### PR TITLE
Make Differential Equations a package extension

### PR DESCRIPTION
**Pull request by tylerhanks**
Making this a draft PR for visibility. I am experiencing an annoying bug where the tests of the DiffEq package extension don't seem to be using the versions of ODEProblem/DDEProblem/etc. overwritten by the package extension file. I'm not sure what is going wrong, but maybe someone else has some idea.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ DifferentialEquations = "0c46a032-eb83-5123-abaf-570d42b7fbaa"
 OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
 
 [extensions]
-AlgebraicDynamicsDiffEqExt = ["DifferentialEquations", "OrdinaryDiffEq", "DelayDiffEq"]
+AlgebraicDynamicsDiffEqExt = ["OrdinaryDiffEq", "DelayDiffEq"]
```
Comment (jpfairbanks) [inline Project.toml]: This should be broken up into 2 extensions one for Ordinary and one for Delay, since those features are orthogonal.

Comment (tylerhanks) [reply Project.toml]: done in most recent commit

**Comment by mehalter** [comment]
Oh also @tylerhanks you probably want to make a commit here as well to add a `codecov.yml` file in the root of the repository with this content:

```yml
comment: false

coverage:
  precision: 2
  status:
    patch:
      default:
        target: 0%
        threshold: 100%
    project:
      default:
        target: 0%
        threshold: 100%
```

This will allow code coverage to execute and provide us information but not fail our github actions if there is a decrease in code coverage

File: Project.toml [hunk 2]
```diff
@@ -7,23 +7,27 @@ version = "0.1.8"
...
-DelayDiffEq = "bcd4f6db-9728-5f36-b5f7-82caef46ccdb"
-DifferentialEquations = "0c46a032-eb83-5123-abaf-570d42b7fbaa"
 LabelledArrays = "2ee39098-c373-598a-b85f-a56591580800"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 LinearMaps = "7a12625a-238d-50fd-b39a-03d52299707e"
-OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
 Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
 RecipesBase = "3cdcf5f2-1ef4-517c-9805-6587b60abb01"
 RecursiveArrayTools = "731186ca-8d62-57ce-b412-fbd966d074cd"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
```
Comment (mehalter) [inline Project.toml:16]:
```suggestion
```

Is `Requires` needed here? This shouldn't be a dependency anymore since we are requiring Julia 1.9. It should also be removed from the `[compat]` section

**Review by mehalter — CHANGES_REQUESTED** [review]

**Review by mehalter — APPROVED** [review]
Looks great!

**Review by jpfairbanks — APPROVED** [review]

File: Project.toml [hunk 3]
```diff
@@ -7,32 +7,33 @@ version = "0.1.8"
 ACSets = "227ef7b5-1206-438b-ac65-934d6da304b8"
```
Comment (mehalter) [inline Project.toml:7]:
```suggestion
```

While we have this PR open, do we need ACSets as a dependency of AlgebraicDynamics? I thought all of these exports were reexported by Catlab already

**Review by mehalter — CHANGES_REQUESTED** [review]

File: Project.toml (hunk 3 above)
Comment (tylerhanks) [reply Project.toml]: ok yeah I didn't realize that if you have Catlab, you don't need ACSets.

File: Project.toml [hunk 4]
```diff
@@ -4,35 +4,35 @@ authors = ["James Fairbanks", "Sophie Libkind"]
...
+DelayDiffEq = "bcd4f6db-9728-5f36-b5f7-82caef46ccdb"
+OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
+
+[extensions]
+AlgebraicDynamicsAlgebraicPetriExt = "AlgebraicPetri"
+AlgebraicDynamicsDelayDiffEqExt = "DelayDiffEq"
+AlgebraicDynamicsOrdinaryDiffEqExt = "OrdinaryDiffEq"
+
 [compat]
 Catlab = "0.15"
```
Comment (mehalter) [inline Project.toml:28]: I think the last thing we are missing is a compat entry for AlgebraicPetri

```suggestion
AlgebraicPetri = "0.9"
Catlab = "0.15"
```

**Review by mehalter — CHANGES_REQUESTED** [review]
Ok, I believe this is the last comment and then we can get this bad boy merged!

**Review by mehalter — APPROVED** [review]
LGTM!

**Comment by mehalter** [comment]
No need to wait for tests to pass this time, latest commit just added a compat entry so I'm going to go ahead and merge this! 🚀
